community/server/src/main/java/org/neo4j/server/rest/transactional/RowWriter.java

```diff
@@ -41,6 +42,25 @@ public void write( JsonGenerator out, Iterable<String> columns, Result.ResultRow
         finally
         {
             out.writeEndArray();
+            writeMeta( out, columns, row );
+        }
+    }
+
+    private void writeMeta( JsonGenerator out, Iterable<String> columns, Result.ResultRow row ) throws IOException
+    {
+        out.writeArrayFieldStart( "meta" );
+        try
+        {
+            // this isn't pretty
```

Comment:
If you are to leave comments like this, maybe it would be better writing _why_ we need to have something that isn't pretty there?